### PR TITLE
Update use of Minimum Outdoor Air Temperature for Compressor Operation

### LINKED ISSUE #6228: Followup to #6158
- unify use of CondenserInletAirTemp between parent and child (both use same information)
- change function call to use int's
- revise input field name? to Minimum Condenser Inlet Air Temperature for Compressor Operation
- VS DX coil example files all use a unity performance curve for Total Cooling Capacity Function of Air Flow Fraction Curve Name. This can lead to RH > 100% issues. A valid curve needs to be developed from manufacturers data (does this even exist) or other source.
- remove minimum outdoor temp for compressor operation field from WSHP and mine information from child object

### LINKED ISSUE #7453: Incorrect variable used in warning when zone air flow is large
Issue overview
--------------
Cut-n-paste error at:
https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/ZoneEquipmentManager.cc#L2659

```
ShowContinueError("...design heating load = " + RoundSigDigits(CalcFinalZoneSizing(I).DesCoolLoad, 2) + " W");
```

In ZoneEquipmentManager::UpdateSizingManager when:

} else if (SELECT_CASE_var == EndZoneSizingCalc) {

Code should read:

```
ShowContinueError("...design heating load = " + RoundSigDigits(CalcFinalZoneSizing(I).DesHeatLoad, 2) + " W");
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
Follow up to #6158 (which fixed Issue #6149) where Minimum Outdoor Air Temperature for Compressor Operation was added to all DX coils. After review it appeared that the controls for systems and coils did not always use the same information.

Fixes #6228 when coil minimum outdoor air temperature for compressor operation was not consistent across system types. Also updates coil documents to include allowed use in AirloopHVAC:UnitarySystem where appropriate.

Fixes #7453 where the wrong variable is used in a high airflow rate sizing warning:
"UpdateZoneSizing: Heating supply air temperature (calculated) within 5/2C of zone temperature" where the cooling load is reported instead of the heating load.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
CI on Win64 diff's:

Warmup convergence diff's:
AirflowNetwork_Multizone_HorizontalOpening.idf

Warmup convergence diff's and Surface View Factor Check Value diff's (unrelated):
StackedZonesWithInterzoneIRTLayers.idf

**Comment by Myoldmopar** [comment]
This is perfect wrt CI.  I'll do some code review and testing soon and if all is well this can drop in.

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -8781,9 +8781,10 @@ namespace DXCoils {
             // (Outdoor wet-bulb temp from DataEnvironment) + (1.0-EvapCondEffectiveness) * (drybulb - wetbulb)
             CondInletTemp = OutdoorWetBulb + (OutdoorDryBulb - OutdoorWetBulb) * (1.0 - DXCoil(DXCoilNum).EvapCondEffect(Mode));
             CondInletHumRat = PsyWFnTdbTwbPb(CondInletTemp, OutdoorWetBulb, OutdoorPressure);
-            CompAmbTemp = CondInletTemp;
+            CompAmbTemp = OutdoorDryBulb;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DXCoils.cc:8784]: Hmm, should remember this for the coil refactor...

File: src/EnergyPlus/DXCoils.cc [hunk 2]
```diff
@@ -10314,10 +10316,12 @@ namespace DXCoils {
         // Get condenser outdoor node info from DX Heating Coil
         if (DXCoil(DXCoilNum).CondenserInletNodeNum(1) != 0) {
             OutdoorDryBulb = Node(DXCoil(DXCoilNum).CondenserInletNodeNum(1)).Temp;
+            CompAmbTemp = OutdoorDryBulb;
             if (DXCoil(DXCoilNum).CondenserType(Mode) == WaterCooled) {
                 OutdoorHumRat = OutHumRat;
                 OutdoorPressure = OutBaroPress;
                 OutdoorWetBulb = OutWetBulbTemp;
+                CompAmbTemp = OutDryBulbTemp;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DXCoils.cc:10324]: This seems to be the one outlier here, while everyone else is OutdoorDryBulb, this is OutDryBulbTemp.  I'm not sure I understand why that is yet, but I'll keep reading along.

File: src/EnergyPlus/Furnaces.cc [hunk 3]
```diff
@@ -1557,7 +1557,7 @@ namespace Furnaces {
 
             // set minimum outdoor temperature for compressor operation
             SetMinOATCompressor(
-                FurnaceNum, Alphas(1), cCurrentModuleObject, CoolingCoilType, CoolingCoilName, HeatingCoilType, HeatingCoilName, ErrorsFound);
+                FurnaceNum, Alphas(1), cCurrentModuleObject, Furnace(FurnaceNum).CoolingCoilIndex, Furnace(FurnaceNum).HeatingCoilIndex, ErrorsFound);
```
Comment (Myoldmopar) [inline src/EnergyPlus/Furnaces.cc:1560]: Passing index instead of type/name is a nice touch.

File: src/EnergyPlus/HVACMultiSpeedHeatPump.cc [hunk 4]
```diff
@@ -1034,13 +1039,12 @@ namespace HVACMultiSpeedHeatPump {
                 ErrorsFound = true;
             }
 
-            MSHeatPump(MSHPNum).MinOATCompressor = Numbers(1);
+            //MSHeatPump(MSHPNum).MinOATCompressor = Numbers(1); // deprecated, now uses coil MinOAT inputs
```
Comment (Myoldmopar) [inline src/EnergyPlus/HVACMultiSpeedHeatPump.cc:1042]: :+1:

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 5]
```diff
@@ -7073,6 +7073,44 @@ namespace VariableSpeedCoils {
...
+    )
+    {
+
+        // FUNCTION INFORMATION:
+        //       AUTHOR         R. Raustad
+        //       DATE WRITTEN   August 2019
+
+        // PURPOSE OF THIS FUNCTION:
+        // This function looks up the the min oat for the cooling coil compressor and returns it.  If
+        // incorrect coil type or name is given, ErrorsFound is returned as true and value is returned
```
Comment (Myoldmopar) [inline src/EnergyPlus/VariableSpeedCoils.cc:7087]: It's not type or name anymore.

File: src/EnergyPlus/VariableSpeedCoils.cc [hunk 6]
```diff
@@ -7073,6 +7073,44 @@ namespace VariableSpeedCoils {
...
+        // as negative.
+
+        // Return value
+        Real64 MinOAT; // returned min oa temperature of matched coil
+
+        // Obtains and Allocates WatertoAirHP related parameters from input file
+        if ( GetCoilsInputFlag ) { // First time subroutine has been entered
```
Comment (Myoldmopar) [inline src/EnergyPlus/VariableSpeedCoils.cc:7094]: No need for the padding around the if condition contents, that's the old format.  I'm a bit surprised that the CI isn't warning on this.

**Review by Myoldmopar — APPROVED** [review]

File: src/EnergyPlus/DXCoils.cc (hunk 2 above)
Comment (rraustad) [reply src/EnergyPlus/DXCoils.cc:10324]: There are 2 indicators here, one is local environment around the compressor (for e.g., crankcase heater) and the other is condenser inlet fluid temp. I'm not sure I have this completely correct but these changes were a big move in the right direction.

File: src/EnergyPlus/DXCoils.cc [hunk 7]
```diff
@@ -10361,14 +10368,14 @@ namespace DXCoils {
         PLRHeating = 0.0;
         DXCoil(DXCoilNum).HeatingCoilRuntimeFraction = 0.0;
         // Initialize crankcase heater, operates below OAT defined in input deck for HP DX heating coil
-        if (OutdoorDryBulb < DXCoil(DXCoilNum).MaxOATCrankcaseHeater) {
+        if (CompAmbTemp < DXCoil(DXCoilNum).MaxOATCrankcaseHeater) {
```
Comment (rraustad) [inline src/EnergyPlus/DXCoils.cc:10371]: So here I think the change is correct.

File: src/EnergyPlus/DXCoils.cc [hunk 8]
```diff
@@ -10361,14 +10368,14 @@ namespace DXCoils {
...
             CrankcaseHeatingPower = DXCoil(DXCoilNum).CrankcaseHeaterCapacity;
         } else {
             CrankcaseHeatingPower = 0.0;
         }
 
         if ((AirMassFlow > 0.0) && (GetCurrentScheduleValue(DXCoil(DXCoilNum).SchedPtr) > 0.0) && (PartLoadRatio > 0.0) &&
-            OutdoorDryBulb > DXCoil(DXCoilNum).MinOATCompressor) {
+            CompAmbTemp > DXCoil(DXCoilNum).MinOATCompressor) {
```
Comment (rraustad) [inline src/EnergyPlus/DXCoils.cc:10378]: But here I am not exactly sure of which indicator should be used. Should this be looking at the local ambient temp or the water inlet temp? Maybe I got his one wrong (not here, but where I set CompAmbTemp for water cooled), I'm not sure. I think this is only questionable for water cooled.

Comment (rraustad) [reply src/EnergyPlus/DXCoils.cc:10378]: Now you have me second guessing myself on whether this line should be reverted.

File: src/EnergyPlus/DXCoils.cc (hunk 7 above)
Comment (rraustad) [reply src/EnergyPlus/DXCoils.cc]: Turns out this was incorrect. Must compare to condenser inlet temp. Corrected in #7483

File: src/EnergyPlus/DXCoils.cc (hunk 8 above)
Comment (rraustad) [reply src/EnergyPlus/DXCoils.cc:10378]: Turns out this was incorrect. Must compare to condenser inlet temp. Corrected in #7483

**Pull request merged**